appcelerator/titanium_mobile
# feat(ios): add LiveView support to Hyperloop

**JIRA:**
- https://jira.appcelerator.org/browse/TIMOB-28461
- https://jira.appcelerator.org/browse/TIMOB-28508

**Summary:**
- Support running LiveView with Hyperloop.
  * No longer modifies JS files containing Hyperloop requires/imports. _(This was the issue.)_
  * Now injects a `hyperloop.bootstrap.js` providing bindings between native types to JS files.
  * Implemented the same way as Android, which supports LiveView as of Titanium 9.0.0.
- Removed 3 year old native property access migration code.
- Re-added 32-bit support needed by Titanium 9.x.x.
  * This reverts [TIMOB-28334](https://jira.appcelerator.org/browse/TIMOB-28334).
- Updated Hyperloop module version to [7.0.4](https://github.com/appcelerator/hyperloop.next/releases/tag/v7.0.4).
  * Module works with Titanium 9.2.0 and higher.

**Note:**
The following still won't work with LiveView because they require a native rebuild.
- Adding libraries.
- Adding `Hyperloop.defineClass()` calls. _(Because it involves native code generation.)_

---
**Test:**
1. Add this PR's hyperloop module to [hyperloop-examples](https://github.com/appcelerator/hyperloop-examples).
2. Build and run with LiveView: `appc run -p ios --liveview`
3. Go to every window in the example app and verify it works as expected.
4. While LiveView is running, open file: `hyperloop-examples/app/controllers/ios/label.js`
5. Change variable `infoString` from `'We ♥ iOS'` to `'We ♥ LiveView'` and save.
6. Verify app restarts without error.
7. Go to the app's Label window and verify the text has been updated.
8. Do an encrypted build with LiveView: `appc run -p ios -D test --liveview`
9. Go to every window in the example app and verify it works as expected.
